redbot/cogs/mutes/mutes.py

```diff
@@ -1560,43 +1576,31 @@ async def mute_user(
         user: discord.Member,
         until: Optional[datetime] = None,
         reason: Optional[str] = None,
-    ) -> Dict[
-        str, Optional[Union[List[Tuple[discord.abc.GuildChannel, str]], discord.Member, bool, str]]
-    ]:
+    ) -> MuteResponse:
         """
         Handles muting users
         """
         permissions = user.guild_permissions
-        ret: Dict[
-            str,
-            Union[bool, Optional[str], List[Tuple[discord.abc.GuildChannel, str]], discord.Member],
-        ] = {
-            "success": False,
-            "reason": None,
-            "channels": [],
-            "user": user,
-        }
-        # TODO: This typing is ugly and should probably be an object on its own
-        # along with this entire method and some other refactorization
-        # v1.0.0 is meant to look ugly right :')
+        ret: MuteResponse = MuteResponse(success=False, reason=None, user=user)
+
         if permissions.administrator:
-            ret["reason"] = _(MUTE_UNMUTE_ISSUES["is_admin"])
+            ret.reason = _(MUTE_UNMUTE_ISSUES["is_admin"])
```

Comment:
Actually this *is* necessary since the call to translate the string does not actually happen until here with the correct context. The initialized variables can be argued for not needing it and I can't remember the exact reason for why we left it this way and I can't find the discussion on it now. I just tested the change and can confirm that removing the translation `_()` from any of these cause the result to not correctly translate the message.

Reply:
`_()` calls in globals are so that redgettext can collect these strings but they're just passthrough and don't do anything (nor should they since the translations need to happen in a context where the locale is properly set).

`_()` calls in actual code are so that the strings actually get translated but those won't be collected by redgettext since they don't operate on string literals.